lib/node_modules/@stdlib/blas/ext/index-of/lib/main.js

```diff
@@ -0,0 +1,204 @@
+		}
+		throw new TypeError( format( 'invalid argument. Second argument must be either an ndarray or a scalar value. Value: `%s`.', searchElement ) );
+	}
+	// Case: indexOf( x, searchElement, ??? )
+	else if ( nargs < 4 ) { // nargs < 4
+		// Case: indexOf( x, searchElement, fromIndex )
+		if ( isndarrayLike( searchElement ) && isndarrayLike( fromIndex ) ) {
+			// As the operation is performed across all dimensions, `se` & `fromIndex` is assumed to be a zero-dimensional ndarray...
+			return base( x, searchElement, fromIndex );
+		}
+		if ( isNumber( searchElement ) && isNumber( fromIndex ) ) {
+			return base( x, broadcastScalar( searchElement, dt, [], ord ), broadcastScalar( fromIndex, DEFAULT_DTYPE, [], ord ) ); // eslint-disable-line max-len
+		}
+		if ( isndarrayLike( searchElement ) && isNumber( fromIndex ) ) {
+			// As the operation is performed across all dimensions, `se` & `fromIndex` is assumed to be a zero-dimensional ndarray...
+			return base( x, searchElement, broadcastScalar( fromIndex, DEFAULT_DTYPE, [], ord ) ); // eslint-disable-line max-len
+		}
+		if ( isNumber( searchElement ) && isndarrayLike( fromIndex ) ) {
+			return base( x, broadcastScalar( searchElement, dt, [], ord ), fromIndex ); // eslint-disable-line max-len
+		}
+
+		// Case: indexOf( x, searchElement, options )
+		opts = fromIndex;
+
+		// Case: indexOf( x, searchElement_ndarray, opts )
+		if ( isndarrayLike( searchElement ) ) {
+			// When not provided `dims`, the operation is performed across all dimensions and `v` is assumed to be a zero-dimensional ndarray; when `dims` is provided, we need to broadcast `v` to match the shape of the non-core dimensions...
+			if ( hasOwnProp( opts, 'dims' ) ) {
+				searchElement = maybeBroadcastArray( searchElement, nonCoreShape( getShape( x ), opts.dims ) ); // eslint-disable-line max-len
+				fromIndex = broadcastScalar( 0, DEFAULT_DTYPE, nonCoreShape( getShape( x ), opts.dims ), ord ); // eslint-disable-line max-len
+			}
+		}
+		// Case: indexOf( x, searchElement_scalar, opts )
+		else if ( isNumber( searchElement ) ) {
+			if ( hasOwnProp( opts, 'dims' ) ) {
+				sh = nonCoreShape( getShape( x ), opts.dims );
+			} else {
+				sh = [];
+			}
+			searchElement = broadcastScalar( searchElement, dt, sh, getOrder( x ) ); // eslint-disable-line max-len
+			fromIndex = broadcastScalar( 0, DEFAULT_DTYPE, sh, ord ); // eslint-disable-line max-len
+		} else {
+			throw new TypeError( format( 'invalid argument. Second argument must be either an ndarray or a numeric scalar value. Value: `%s`.', searchElement ) );
+		}
+		return base( x, searchElement, fromIndex, opts );
+	}
+
+	// Case: indexOf( x, searchElement, fromIndex, options )
+	opts = options;
+
+	// Case: indexOf( x, searchElement_ndarray, fromIndex_ndarray, opts )
+	if ( isndarrayLike( searchElement ) && isndarrayLike( fromIndex ) ) {
```

Comment:
Same comment. You are doing repeated brand checks. Handle each argument separately. Normalize each to a broadcasted ndarray. Do not do what you are doing here with handling combinations. This just leads to noise and repeated logic.